### PR TITLE
[Issue 643] Column Property addition

### PR DESCRIPTION
**Pull request by ashokblend**
1.Added interface to get dictionary service
2.Added class to identify column
3.Modified schema.thrift to add column properties

### CODE REVIEW

**Comment by ashokblend** [comment]
@gvramana  please review it

File: integration/spark/src/main/scala/org/carbondata/spark/util/GlobalDictionaryUtil.scala [hunk 1]
```diff
@@ -293,52 +302,13 @@ object GlobalDictionaryUtil extends Logging {
...
-            throw new IOException(s"Failed to created dictionary folder: ${dictfolderPath}")
-        }
-        primDimensions.zipWithIndex.foreach { f =>
-          dictFilePaths(f._2) = carbonTablePath.getDictionaryFilePath(f._1.getColumnId)
-          // all dictionary files are not exists
-          dictFileExists(f._2) = false
-        }
-    }
+    val primDimensions = primDimensionsBuffer.map { x => x }.toArray
+    val dictDetail = CarbonSparkFactory.getDictionaryDetailService().
```
Comment (ashokblend) [inline integration/spark/src/main/scala/org/carbondata/spark/util/GlobalDictionaryUtil.scala]: @QiangCai please review it

File: integration/spark/src/main/scala/org/carbondata/spark/DictionaryDetailHelper.scala [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import org.carbondata.core.carbon.path.CarbonStorePath
+import org.carbondata.core.carbon.path.CarbonTablePath
+import org.carbondata.core.datastorage.store.filesystem.CarbonFile
+import org.carbondata.core.datastorage.store.filesystem.CarbonFileFilter
+import org.carbondata.core.datastorage.store.impl.FileFactory
+
+class DictionaryDetailHelper extends DictionaryDetailService {
+  def getDictionaryDetail(dictfolderPath: String, primDimensions: Array[CarbonDimension],
+      table: CarbonTableIdentifier, hdfsLocation: String): DictionaryDetail = {
+    val carbonTablePath = CarbonStorePath.getCarbonTablePath(hdfsLocation, table)
```
Comment (ashokblend) [inline integration/spark/src/main/scala/org/carbondata/spark/DictionaryDetailHelper.scala]: @QiangCai  please review this class

File: integration/spark/src/main/scala/org/carbondata/spark/DictionaryDetailHelper.scala [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    // 2 put dictionary file names to fileNamesMap
+    val fileNamesMap = new HashMap[String, Int]
+    for (i <- 0 until carbonFiles.length) {
+      fileNamesMap.put(carbonFiles(i).getName, i)
+    }
+    // 3 lookup fileNamesMap, if file name is in fileNamesMap, file is exists, or not.
+    primDimensions.zipWithIndex.foreach { f =>
+      columnIdentifier(f._2) = f._1.getColumnIdentifier
+      dictFilePaths(f._2) = carbonTablePath.getDictionaryFilePath(f._1.getColumnId)
+      dictFileExists(f._2) =
```
Comment (QiangCai) [inline integration/spark/src/main/scala/org/carbondata/spark/DictionaryDetailHelper.scala]: please check whether dictFilePaths(f._2) is under dictfolderPath or not.

File: core/src/main/java/org/carbondata/core/writer/CarbonDictionaryWriterImpl.java [hunk 4]
```diff
@@ -248,6 +247,14 @@ private void init() throws IOException {
     createChunkList();
   }
 
+  protected void initPaths() {
+    CarbonTablePath carbonTablePath =
```
Comment (gvramana) [inline core/src/main/java/org/carbondata/core/writer/CarbonDictionaryWriterImpl.java]: we need to use PathService here

File: integration/spark/src/main/scala/org/apache/spark/sql/execution/command/carbonTableSchema.scala [hunk 5]
```diff
@@ -1966,38 +1982,38 @@ private[sql] case class DescribeCommandFormatted(
...
-        relation.cubeMeta.carbonTableIdentifier.getTableName).asScala.toList)
-    results ++= Seq(("", "", ""), ("#Aggregate Tables", "", ""))
+    results ++= Seq(("Table Name : ", relation.cubeMeta.carbonTableIdentifier.getTableName, "", ""))
+    results ++= Seq(("CARBON Store Path : ", relation.cubeMeta.storePath, "", ""))
+    results ++= Seq(("", "", "", ""), ("#Aggregate Tables", "", "", ""))
     val carbonTable = relation.cubeMeta.carbonTable
     val aggTables = carbonTable.getAggregateTablesName
     if (aggTables.size == 0) {
-      results ++= Seq(("NONE", "", ""))
+      results ++= Seq(("NONE", "", "", ""))
```
Comment (gvramana) [inline integration/spark/src/main/scala/org/apache/spark/sql/execution/command/carbonTableSchema.scala]: Dont show all the table properties at end of description, but not in each column. Column width will not be cleanly readable in shell.

File: integration/spark/src/main/scala/org/carbondata/spark/DictionaryDetailHelper.scala (hunk 3 above)
Comment (ashokblend) [inline integration/spark/src/main/scala/org/carbondata/spark/DictionaryDetailHelper.scala]: This scenario will not occur and hence its not required.

**Comment by ashokblend** [comment]
@gvramana  all comments are handled, please check.

File: core/src/main/java/org/carbondata/core/writer/CarbonDictionaryWriterImpl.java (hunk 4 above)
Comment (ashokblend) [inline core/src/main/java/org/carbondata/core/writer/CarbonDictionaryWriterImpl.java]: It's handled

File: integration/spark/src/main/scala/org/apache/spark/sql/execution/command/carbonTableSchema.scala (hunk 5 above)
Comment (ashokblend) [inline integration/spark/src/main/scala/org/apache/spark/sql/execution/command/carbonTableSchema.scala]: It's handled

**Comment by gvramana** [comment]
LGTM

**Pull request merged**